dcrutil/wif.go

```diff
@@ -10,19 +10,24 @@ import (
 	"errors"
 
 	"github.com/decred/base58"
-	"github.com/decred/dcrd/chaincfg"
-	"github.com/decred/dcrd/chaincfg/chainec"
 	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v2/chainec"
 	"github.com/decred/dcrd/dcrec"
 	"github.com/decred/dcrd/dcrec/edwards"
 	"github.com/decred/dcrd/dcrec/secp256k1"
 )
 
-// ErrMalformedPrivateKey describes an error where a WIF-encoded private
-// key cannot be decoded due to being improperly formatted.  This may occur
-// if the byte length is incorrect or an unexpected magic number was
-// encountered.
-var ErrMalformedPrivateKey = errors.New("malformed private key")
+var (
+	// ErrMalformedPrivateKey describes an error where a WIF-encoded private key
+	// cannot be decoded due to being improperly formatted.  This may occur if
+	// the byte length is incorrect or an unexpected magic number was
+	// encountered.
+	ErrMalformedPrivateKey = errors.New("malformed private key")
+
+	// ErrWrongWIFNetwork desribes an error in which the provided WIF is not for
+	// the expected network.
+	ErrWrongWIFNetwork = errors.New("the provided WIF is for the wrong network")
```

Comment:
or "not for the %s network"?

I would be fine with making this a fmt.Errorf with that description, or adding some generic ErrUnknownNetwork for any unexpected magic for both addresses and wifs and anything else we may encounter.

Reply:
It can only provide the magic bytes instead of a nice network name since that's all that is available, but I can update it with that.

Reply:
dynamically check for a Stringer and fall back to more generic text otherwise?

Reply:
Done.  Example error:
`WIF is not for the network identified by 0xabcd`

Reply:
The API just takes `[2]byte` for the expected network magics.

Reply:
Done.